operatorPointwise_binary.go

```diff
@@ -422,26 +409,14 @@ func hadamardProdDiff(x, y, z *Node) (err error) {
 	mul := newEBOByType(mulOpType, ydv.Value.Type(), zdvdType)
 	err = mul.IncrDo(xdv.d, ydv.Value, zdv.d)
 	if err != nil {
-		var ver Valuer
-		var ok bool
-		if ver, ok = err.(Valuer); !ok {
-			return
-		}
-
-		xdv.SetDeriv(ver.Value()) // ignore errors on purpose
+		return errors.Wrap(err, "IncrDo call failed")
```

Comment:
I have noticed that my scalar*tensor and tensor*scalar ops are now failing... I will revert this.